tests/integration/golang/aim/metric/search_test.go

```diff
@@ -34,97 +41,748 @@ func (s *SearchMetricsTestSuite) SetupTest() {
+			request: request.SearchMetricRequest{
+				Query: `((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and metric.last >= 1.6 and run.name != "TestRun1"`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric2Run2,
+				metric3Run2,
+				metric2Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastRunNameOperationLessOrEqualAndNotEqual",
+			request: request.SearchMetricRequest{
+				Query: `((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and metric.last <= 1.6 and run.name != "TestRun1"`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run2,
+				metric1Run3,
+				metric2Run3,
+			},
+		},
+		{
+			name: "SearchMetricComplexQuery",
+			request: request.SearchMetricRequest{
+				Query: `((metric.name == "TestMetric1") or (metric.name == "TestMetric2")) and metric.last_step >= 1 and (run.name.endswith("2") or re.match("TestRun1", run.name)) and (metric.last < 1.6) and run.duration > 0`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run2,
+			},
 		},
 	}
 	for _, tt := range tests {
 		s.T().Run(tt.name, func(T *testing.T) {
-			data, err := s.client.DoStreamRequest(
+			var resp []byte
+			query, err := urlquery.Marshal(tt.request)
+			assert.Nil(s.T(), err)
+			resp, err = s.client.DoStreamRequest(
 				http.MethodGet,
-				fmt.Sprintf("/runs/search/metric?%s", tt.query),
+				fmt.Sprintf(`/runs/search/metric?%s`, query),
 				nil,
 			)
-			decodedData, err := encoding.Decode(bytes.NewBuffer(data))
 			assert.Nil(s.T(), err)
-			value, ok := decodedData["id.props.name"]
-			assert.True(s.T(), ok)
-			assert.Equal(s.T(), "chill-run", value)
+			decodedData, err := encoding.Decode(bytes.NewBuffer(resp))
+			assert.Nil(s.T(), err)
+
+			decodedMetrics := []*models.LatestMetric{}
+			for _, run := range runs {
+				metricCount := 0
+				for decodedData[fmt.Sprintf("%v.traces.%d.name", run.ID, metricCount)] != nil {
+					epochsKey := fmt.Sprintf("%v.traces.%d.epochs.blob", run.ID, metricCount)
+					itersKey := fmt.Sprintf("%v.traces.%d.iters.blob", run.ID, metricCount)
+					nameKey := fmt.Sprintf("%v.traces.%d.name", run.ID, metricCount)
+					timestampsKey := fmt.Sprintf("%v.traces.%d.timestamps.blob", run.ID, metricCount)
+					valuesKey := fmt.Sprintf("%v.traces.%d.values.blob", run.ID, metricCount)
+
+					m := models.LatestMetric{
+						Key:       decodedData[nameKey].(string),
+						Value:     decodedData[valuesKey].([]float64)[0],
+						Timestamp: int64(decodedData[timestampsKey].([]float64)[0] * 1000),
+						Step:      int64(decodedData[epochsKey].([]float64)[0]),
+						IsNan:     false,
+						RunID:     run.ID,
+						LastIter:  int64(decodedData[itersKey].([]float64)[0]),
+					}
+					decodedMetrics = append(decodedMetrics, &m)
+					metricCount++
+				}
+			}
+
+			// Check if the received metrics match the expected ones
+			for _, metric := range metrics {
```

Comment:
can't understand why we can't just compare `tt.metrics` and `decodedMetrics`? can you please explain why do we need `metrics`?

Reply:
You're right, I was doing this to check if the not expected ones weren't in the response but yes we can check it by comparing `tt.metrics` and `decodedMetrics`, I'll fix this, thanks!

Reply:
but you can create a helper function to apply smart comparing: length is equal, all elements existing in both slices, etc.